services/src/main/java/org/keycloak/exportimport/util/ExportUtils.java

```diff
@@ -267,6 +264,21 @@ public static RealmRepresentation exportRealm(KeycloakSession session, RealmMode
                 rep.setFederatedUsers(federatedUsers);
             }
 
+        } else if (options.isClientsIncluded() && options.isOnlyServiceAccountsIncluded()) {
+            List<UserRepresentation> users = new LinkedList<>();
+            for (ClientModel app : clients) {
+                if (app.isServiceAccountsEnabled()) {
```

Comment:
I think we also could check if the client is public. Let's say, someone configures a confidential client, enables service account but then decides to switch the client back to public (which effectively prevents the service account from logging in but doesn't disable it). In this case the service account is still exported.

Reply:
True, I have added that the client is not public and not bearer to the if.